Update customization_dev.groovy

gergoe opened the pull request:
Ensure a File property is not checked against isInteger() when setting preferences.

gergoe committed:
Update customization_dev.groovy

gergoe committed:
Revert "Update customization_dev.groovy"

This reverts commit e0ef749168edf7904f7ad41573ee92953a41d8db.

gergoe committed:
Update customization_dev.groovy

gergoe commented:
Sorry for the extra commits, overlooked something.

briacp commented on scripts/application_startup/customization_dev.groovy:
```diff
@@ -335,7 +335,9 @@ void update_omegat_prefs(localPrefsPath) {
     // Try to guess the property type, Boolean, Integer or String
     remotePrefs.each { prop ->
-        if (prop.value == "true" || prop.value == "false") {
+        if (prop.value instanceof File) {
```
```
        if (prop.value instanceof File) {
```

Do you know where a File object is set as a Preference? I can't find any reference.

In any case, it would probably be better to do the following (to catch other cases)

```java
        } else {
            Preferences.setPreference(prop.key, prop.value.toString())
        }
```

gergoe replied:
```Groovy
local_scripts_dpath = new File(config_dir.toString() + File.separator + "scripts")
```

and later inside update_omegat_prefs()
```Groovy
    remotePrefs.scripts_dir = local_scripts_dpath
```

In any case, it might be better indeed to fix it right there, by changing it into the following:
```Groovy
    remotePrefs.scripts_dir = local_scripts_dpath.toString()
```


Speaking of which; what's the purpose of forcing integer looking strings to be converted to Integer, does it make a difference somewhere? If not then your suggestion is indeed better.

briacp replied:
> Speaking of which; what's the purpose of forcing integer looking strings to be converted to Integer, does it make a difference somewhere? If not then your suggestion is indeed better.

Inside the code of OmegaT, few preferences are directly set as Integer/Boolean (and not parsed from a String like most other prefs). If we try to set these preferences with a String, it will throw a ClassCastException

gergoe committed:
Update customization_dev.groovy

gergoe commented:
Fixed the File class issue where it is caused, reverted extra if conditions.

briacp commented on scripts/application_startup/customization_dev.groovy:
```diff
@@ -330,14 +330,12 @@ void update_omegat_prefs(localPrefsPath) {
 
     console.println(" I will set scripts_dir to '${local_scripts_dpath}'")
-    remotePrefs.scripts_dir = local_scripts_dpath
+    remotePrefs.scripts_dir = local_scripts_dpath.toString()
```
While Groovy can implicitly set map entries this way, I find it a bit confusing when reading the code afterward; also `getAbsolutePath()` makes the intent clearer (and it's a String too!)

```suggestion
    remotePrefs.put('scripts_dir', local_scripts_dpath.getAbsolutePath())
```

gergoe replied:
Totally agree.

gergoe committed:
Update scripts/application_startup/customization_dev.groovy

Co-authored-by: Briac Pilpré <briacp@gmail.com>

gergoe committed:
Merging changes of customization_dev.groovy

Merging the changes of the two copies of customization_dev.groovy so they are the same now.
